pkg/types/validation/machinepools.go

```diff
@@ -61,6 +61,9 @@ func ValidateMachinePool(platform *types.Platform, p *types.MachinePool, fldPath
 	if !validArchitectures[p.Architecture] {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("architecture"), p.Architecture, validArchitectureValues))
 	}
+	if p.Platform.Azure != nil && p.Name == masterPoolName && p.Platform.Azure.OSDisk.DiskType == "Standard_LRS" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("diskType"), p.Platform.Azure.OSDisk.DiskType, "DiskType not compatible with master machines."))
```

Comment:
`master machines` -> `control plane`
Since this is validating the default machine pool perhaps the error message could point the user that this is a valid selection for the control plane machine pool